infobind/cache.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infobind
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/terror"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/sqlexec"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultBindCacheSize = 5
+
+// BindData store the basic bind info and bindSql astNode.
+type BindData struct {
+	bindRecord
+	ast ast.StmtNode
+}
+
+// BindCache hold a bindDataMap, key:origin sql hash value: bindData slice.
+type BindCache struct {
+	Cache map[string][]*BindData
+}
+
+// Handle hold a atomic bindCache.
+type Handle struct {
+	bind atomic.Value
+}
+
+// HandleUpdater use to update the BindCache.
+type HandleUpdater struct {
+	Parser         *parser.Parser
+	LastUpdateTime types.Time
+	Ctx            sessionctx.Context
```

Comment:
Why export them?

Reply:
We need initial these varibies in domain.

Reply:
Then just use a function to init them.

Reply:
ok